openfl/utilities/attestation/attestation_utils.py

```diff
@@ -0,0 +1,259 @@
+
+from base64 import b64encode
+import json
+import base64
+import os
+import http
+import requests
+import hashlib
+import logging
+
+from openfl.utilities.attestation.signer_utils import ECDSASigner
+
+
+logger = logging.getLogger(__name__)
+
+class AttestationManager:
+    """Class to manage attestation for enclaves.
+    This class handles the generation of SGX quotes, fetching MRENCLAVE values,
+    and obtaining attestation reports from AVS (Attestation Verification Service).
+    """
+    
+
+    def __init__(self, participant_name, attestation_report_path, ita_api_key, avs_url):
+        """Initializes the AttestationManager with the provided parameters.
+
+        Args:
+            participant_name (str): Name of the enclave.
+            attestation_report_path (str): Path to the store attestation reports.
+            ita_api_key (str): API key for ITA.
+            avs_url (str): URL for AVS.
+            enable_remote_attestation (bool): Flag to enable remote attestation.
+        """
+        self.participant_name = participant_name
+        self.attestation_report_path = attestation_report_path
+        self.ita_api_key = ita_api_key
+        self.avs_url = avs_url
+
+
+    def get_attestation_report(
```

Comment:
as this also constructs the enclave's identity (ECDSA key pair), I suggest renaming this method to `get_attested_identity`

Reply:
renamed